### PR TITLE
feat: add LNURL verify

### LINKED ISSUE #2: support LNURL-verify
This should be simple because we the verify URL can use just the lookup_invoice method using the payment hash which is available from calling make_invoice when doing the lnurlp callback.

### PR DESCRIPTION
**Pull request by im-adithya**
## Description

Fixes #2

- Adds LNURL verify at `/lnurlp/:username/verify/:payment_hash`
- The callback URL is shifted from `/.well-known/lnurlp/:username/callback` to `/lnurlp/:username/callback`
- Context type is added to avoid ts errors
- DEBUG is capitalized as `LevelName` type is defined as `type LevelName = "NOTSET" | "DEBUG" | "INFO" | "WARN" | "ERROR" | "CRITICAL"`

### CODE REVIEW

File: src/lnurlp.ts [hunk 1]
```diff
@@ -52,17 +58,45 @@ export function createLnurlApp(db: DB) {
       });
 
       const transaction = await nwcClient.makeInvoice({
-        amount: +amount,
+        amount: Math.floor(+amount / 1000) * 1000,
```
Comment (rolznz) [inline src/lnurlp.ts:61]: Why was this added? you cannot have sub-millisats

File: src/lnurlp.ts [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
 import { BASE_URL, DOMAIN } from "./constants.ts";
 import { DB } from "./db/db.ts";
 import "./nwc/nwcPool.ts";
 
-export function createLnurlApp(db: DB) {
+export function createLnurlWellKnownApp(db: DB) {
   const hono = new Hono();
 
-  hono.get("/:username", async (c) => {
+  hono.get("/:username", async (c: Context) => {
```
Comment (rolznz) [inline src/lnurlp.ts:11]: You should not need `: Context`. In Typescript, it is inferred from the argument type. Are you sure your environment is setup correctly?

File: src/lnurlp.ts [hunk 3]
```diff
@@ -52,17 +58,45 @@ export function createLnurlApp(db: DB) {
...
+        nostrWalletConnectUrl: connectionSecret,
+      });
+
+      const transaction = await nwcClient.lookupInvoice({
+        payment_hash: paymentHash,
+      });
+
+      return c.json({
+        settled: !!transaction.settled_at,
+        preimage: transaction.preimage || null,
```
Comment (rolznz) [inline src/lnurlp.ts:93]: Why `|| null`? shouldn't it be omitted from the response body if it is undefined?

Comment (bumi) [reply src/lnurlp.ts:93]: not sure what the spec says, but having the key in the response but a blank value makes sense to me.

**Comment by bumi** [comment]
lgtm

File: src/lnurlp.ts (hunk 1 above)
Comment (im-adithya) [reply src/lnurlp.ts:61]: Because if someone mistakes and sends the amount param as sats instead of millisats we should be able to round it off, currently it results in a weird invoice which shows the amount as 0 sats in the extension when trying to pay. `https://getalby.com/lnurlp/adithya/callback` also rounds the amount

File: src/lnurlp.ts (hunk 3 above)
Comment (im-adithya) [reply src/lnurlp.ts:93]: +1 I followed the rails app implementation, better to return `null` right?

Comment (bumi) [reply src/lnurlp.ts:93]: I assume in JSON undefined and null end up equal?

Comment (rolznz) [reply src/lnurlp.ts:93]: image

They are not the same. Null is supposedly a valid json value now. Not all parsers support it (but most do). I guess it's ok. I will resolve this one

File: src/lnurlp.ts (hunk 1 above)
Comment (rolznz) [reply src/lnurlp.ts:61]: Thanks for checking!

**Review by rolznz — APPROVED** [review]
Nice branch name :laughing: 

utACK
